Sources/Subprocess/Configuration.swift

```diff
@@ -640,84 +649,248 @@ internal struct TrackedFileDescriptor: ~Copyable {
             // Throw other kinds of errors to allow user to catch them
             throw error
         }
+    case .dispatchIO(let dispatchIO):
+        dispatchIO.close()
+    }
+}
+
+/// `IODescriptor` wraps platform-specific `FileDescriptor`,
+/// which is used to establish a connection to the standard input/output (IO)
+/// system during the process of spawning a child process. Unlike `IODescriptor`,
+/// the `IODescriptor` does not support data read/write operations;
+/// its primary function is to facilitate the spawning of child processes
+/// by providing a platform-specific file descriptor.
+internal struct IODescriptor: ~Copyable {
+    #if canImport(WinSDK)
+    typealias Descriptor = HANDLE
+    #else
+    typealias Descriptor = FileDescriptor
+    #endif
+
+    internal var closeWhenDone: Bool
+    internal let descriptor: Descriptor
+
+    internal init(
+        _ descriptor: Descriptor,
+        closeWhenDone: Bool
+    ) {
+        self.descriptor = descriptor
+        self.closeWhenDone = closeWhenDone
+    }
+
+    consuming func createIOChannel() -> IOChannel {
+        let shouldClose = self.closeWhenDone
+        self.closeWhenDone = false
+        #if canImport(Darwin)
+        // Transferring out the ownership of fileDescriptor means we don't have go close here
+        let closeFd = self.descriptor
+        let dispatchIO: DispatchIO = DispatchIO(
+            type: .stream,
+            fileDescriptor: self.platformDescriptor(),
+            queue: .global(),
+            cleanupHandler: { error in
```

Comment:
I think you made some additional changes since I mentioned this in https://github.com/swiftlang/swift-subprocess/pull/64/files#r2184424762, so I'd suggest adding the `@Sendable` annotation here (since from my reading it looks like it should work now) to avoid any potential build errors if that annotation is corrected at some point.

```suggestion
            cleanupHandler: { @Sendable error in
```